enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,280 @@
+```
+# HELP cluster_installation_invocation represents a single command run by the OpenShift installer and related outcomes.
+# TYPE cluster_installation_invocation counter
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="25"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="22.5"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="APIFailure",return="1",version="4.2",duration="30"} 1
+cluster_installation_invocation{command="destroy", target="cluster",os="linux",platform="AWS",result="Success",return="1",version="4.2",duration="10"} 1
+```
+The `result` label denotes a command outcome:
+* `ProvisioningFailed` - for `create` the installer failed to provision resources
+* `BootstrapFailed` - for `create` the bootstrap node failed
+* `APIFailed` - for `create` and `wait-for` the control plane API never came up
+* `OperatorsDegraded` - for `create` and `wait-for` command not all operators became available
+* `Success` - for `create` and `wait-for` the cluster completed successfully and other commands completed successfully
+
+TODO: determine relevant results for other commands, perhaps inability to connect to SSH with `destroy` or `gather` 
+
+#### cluster_installation_duration
+The cluster_installation_duration metric is pushed after execution of the `create` and `wait-for` commands. The duration of each stage of installation is similar to duration in the previous metric.
+
+```
+# HELP cluster_installation_duration keeps track of the count of the duration of individual stages of the command execution in buckets with the result of the command.
+# TYPE cluster_installation_duration counter
+cluster_installation_duration{provisioning="5", infrastructure="5", bootstrap="5", api="5", operators="5" success="true"} 1
+```
+One sample would be pushed per command invocation.
+
+#### cluster_installation_modification
+The `cluster_installation_modification` metric represents whether a user modified an installer created asset prior to running the `create` command.
+
+Feedback would be appreciated on two potential designs. One which aggregates the values into a single sample and one that pushes multiple samples:
+
+```
+# HELP cluster_installation_modification lists all the assets that were changed from the original value by the user before the create command was called.
+# TYPE cluster_installation_modification counter
+cluster_installation_modification{asset="cluster-config.yaml,cluster-scheduler-02-config.yml"} 1
```

Comment:
Agreed, the upper bound of label values must be known in forehand.

Reply:
I have updated the enhancement to provide more context. I think we have potentially ~30 values, with a reasonable expectation it could be much lower. Let me know if that sounds like a reasonable bound.